### ISSUE TITLE
Firefox 72: Pausing in Picture-in-picture mode unpauses immediately

### ISSUE THREAD

**Issue by Naryosha** [lawfx/YoutubeNonStop#22]
Browser: Mozilla Firefox 72.0.1 (64-bit)
OS: Ubuntu 19.10 (eoan)

This pause button on Picture-in-Picture overlay unpauses immediately upon clicking. Can pause normally after disabling the extension.
![image](https://user-images.githubusercontent.com/43379543/72270464-c8f86c80-3657-11ea-9cc2-14fee2ce0616.png)

https://support.mozilla.org/en-US/kb/about-picture-picture-firefox

**Comment by jacekmusial** [comment]
Extension version: 0.7.1
Browser: freshly downloaded Mozilla Firefox 72.0.1 (64 bit) with only one extension YoutubeNonStop
OS: Ubuntu 18.04.3 LTS

I can not reproduce it on Youtube website

![image](https://user-images.githubusercontent.com/7474972/72532410-4516ce00-386b-11ea-9b66-4591bb53a1da.png)


![Peek 2020-01-16 14-28](https://user-images.githubusercontent.com/7474972/72533089-7348dd80-386c-11ea-93b1-cd6170a9ef23.gif)

**Comment by Naryosha** [comment]
@jacekmusial 
I think that's Youtube Mini Player (https://support.google.com/youtube/answer/9162927?hl=en) which is Youtube's feature and unaffected. The issue affects Picture-in-picture mode which is Firefox's feature. One of the differences is that Picture-in-picture allows you to minimize the browser and have the video stays on top.

**Comment by jacekmusial** [comment]
You are right, watching any video in Firefox picture-in-picture mode I was able to reproduce it. After enabling the plugin, issue comes up: after pausing the video it unpauses itself a moment later.

**Comment by RedEyedRocker** [comment]
@lawfx Please take a look at this issue, it is very annoying for users that use Firefox's Picture-in-Picture mode.

**Comment by lawfx** [comment]
This issue is the same issue as #18 and a couple of others opened in the past.
The general problem is that when you act on youtube outside of the youtube's page, YTNS has no information on whether it was a user action or something else, and that's why it's unpausing the video.

I haven't found a way to fix that yet, though a temporary solution is:
1. Click "Pause" media key
2. YouTube NonStop resumes video
3. Immediately click "Pause" media key again
4. Video remains paused

**Comment by RedEyedRocker** [comment]
> 
> 
> This issue is the same issue as #18 and a couple of others opened in the past.
> The general problem is that when you act on youtube outside of the youtube's page, YTNS has no information on whether it was a user action or something else, and that's why it's unpausing the video.
> 
> I haven't found a way to fix that yet, though a temporary solution is:
> 
>     1. Pause the video
> 
>     2. YouTube NonStop resumes video
> 
>     3. Immediately pause again(you have to be kinda fast)
> 
>     4. Video remains paused

But why don't you make it so the addon simply only unpauses when it's the Youtube's script that pauses the video due to inactivity? That way, it won't interfere with manual pausing.

**Comment by lawfx** [comment]
@RedEyedRocker because I don't know how to detect if it's the YouTube's code that does something. It works by detecting if the video goes in paused mode. If you have an idea how to track down the action from YouTube that causes the pausing, let me know.

**Comment by lawfx** [comment]
I posted an update for chrome with support for media keys, PiP mode and all that. In Firefox it will come in the next few days if everything goes well.

**Comment by AGonLen** [comment]
If it helps, I have the same, Firefox 76.0.1 and media.hardwaremediakeys.enabled  and dom.media.mediasession.enabled  are both as "False" and Play/Pause PiP button is there for Firefox, but as you know, not working when YoutubeNonStop is enabled.

**Comment by Linard11** [comment]
The issue persists on Firefox 82. One thing I did notice is that the problem only occurs if the media controls on the pop-out window have dissapeared at least once. If you are unpausing a video and keep the overlay alive, you can pause/unpause at will. Only if you let the overlay time out and then try to pause/unpause it stops working with YTNS enabled.
